### DISCUSSION TITLE
/scrape API - custom behavior with waitFor and dynamical loading page

### DISCUSSION THREAD

**Discussion by macrochel** [browserless/chrome#3871] [Q&A] [answered]
I need to make scrape the assortment, but page is loading dynamically. I need to create a function, that will wait when every loadings will finish (on this page there's not just one load)

`{
  "url": "https://a201910091655256231033.szwego.com/static/index.html#/shop_detail/A201910091655256231033",
  "elements": [{"selector": ".word-break, .lazy"}],
  "waitFor": 10000
}`

Now I can get only 128 items

**Comment by joelgriffith** [comment, marked as answer]
Have you tried the function endpoint? That's more robust and allows you to programmatically write your own code to run: https://docs.browserless.io/open-api#tag/Browser-REST-APIs/paths/~1pdf%20~1chromium~1pdf/post
